dask/array/tests/test_overlap.py

```diff
@@ -599,13 +599,34 @@ def test_depth_greater_than_boundary_length():
     assert_array_equal(result, expected)
 
 
-def test_bad_depth_raises():
-    expected = np.arange(144).reshape(12, 12)
-    darr = da.from_array(expected, chunks=(5, 5))
+def test_depth_greater_than_smallest_chunk_combines_chunks():
+    a = np.arange(144).reshape(12, 12)
+    darr = da.from_array(a, chunks=(5, 5))
+    assert darr.chunks == ((5, 5, 2), (5, 5, 2))
+
+    depth = {0: 4, 1: 2}
+    output = overlap(darr, depth=depth, boundary=1)
+    assert output.chunks == ((2 * 4 + 5, 2 * 4 + 7), (2 * 2 + 5, 2 * 2 + 5, 2 * 2 + 2))
+
+
+def test_depth_greater_than_several_chunks_rechunks():
+    a = np.arange(144).reshape(12, 12)
+    darr = da.from_array(a, chunks=(3, 11))
+    assert darr.chunks == ((3, 3, 3, 3), (11, 1))
 
     depth = {0: 4, 1: 2}
+    output = overlap(darr, depth=depth, boundary=1)
+    assert output.chunks == ((2 * 4 + 6, 2 * 4 + 6), (2 * 2 + 12,))
```

Comment:
Combining chunks seems clean, but I could see an argument for these to be rechunked into (prior to padding): 
(4,4,4) and (10, 2)

Reply:
I think I'd rather not get into that kind of rechunking since that is very easy for people to do their own selves.